sdk/search/azure-search/src/test/java/com/azure/search/LookupAsyncTests.java

```diff
@@ -281,9 +267,90 @@ public void emptyDynamicObjectsInCollectionExpandedOnGetWhenCollectionFieldSelec
         uploadDocument(client, originalDoc);
         List<String> selectedFields = Arrays.asList("HotelId", "Rooms");
 
-        Mono<Document> result = client.getDocument("1", selectedFields, generateRequestOptions());
-        StepVerifier.create(result)
-            .assertNext(actualDoc -> Assert.assertEquals(expectedDoc, actualDoc))
+        StepVerifier.create(client.getDocumentWithResponse("1", selectedFields, generateRequestOptions()))
+            .assertNext(response -> assertEquals(expectedDoc, response.getValue()))
+            .verifyComplete();
+    }
+
+    @Override
+    public void getDynamicDocumentCannotAlwaysDetermineCorrectType() {
+        createHotelIndex();
+        client = getSearchIndexClientBuilder(INDEX_NAME).buildAsyncClient();
+
+        List<Document> rooms = new ArrayList<>();
+        rooms.add(new Document(Collections.singletonMap("baseRate", NaN)));
+
+        Document indexedDoc = new Document();
+        indexedDoc.put("HotelId", "1");
+        indexedDoc.put("HotelName", "2015-02-11T12:58:00Z");
+        indexedDoc.put("Location", GeoPoint.create(40.760586, -73.975403)); // Test that we don't confuse Geo-JSON & complex types.
+        indexedDoc.put("Rooms", rooms);
+
+        Document expectedDoc = new Document();
+        expectedDoc.put("HotelId", "1");
+        expectedDoc.put("HotelName", OffsetDateTime.of(2015, 2, 11, 12, 58, 0, 9, ZoneOffset.UTC));
+        expectedDoc.put("Location", GeoPoint.create(40.760586, -73.975403));
+        expectedDoc.put("Rooms", Collections.singleton(new Document(Collections.singletonMap("BaseRate", "NaN"))));
+
+        IndexBatch<Document> batch = new IndexBatch<>();
+        batch.addUploadAction(expectedDoc);
```

Comment:
Looks like the bug from the Track 1 tests got ported over. This should be uploading `indexedDoc`, not `expectedDoc`.

Reply:
Removed this class